packages/mdc-slider/adapter.js

```diff
@@ -0,0 +1,192 @@
+   */
+  computeBoundingRect() {}
+
+  /**
+   * Returns the tab index of the slider Element
+   * @return {number}
+   */
+  getTabIndex() {}
+
+  /**
+   * Registers an event handler on the root element for a given event.
+   * @param {string} type
+   * @param {function(!Event): undefined} handler
+   */
+  registerInteractionHandler(type, handler) {}
+
+  /**
+   * Deregisters an event handler on the root element for a given event.
+   * @param {string} type
+   * @param {function(!Event): undefined} handler
+   */
+  deregisterInteractionHandler(type, handler) {}
+
+  /**
+   * Registers an event handler on the thumb container element for a given event.
+   * @param {string} type
+   * @param {function(!Event): undefined} handler
+   */
+  registerThumbContainerInteractionHandler(type, handler) {}
+
+  /**
+   * Deregisters an event handler on the thumb container element for a given event.
+   * @param {string} type
+   * @param {function(!Event): undefined} handler
+   */
+  deregisterThumbContainerInteractionHandler(type, handler) {}
+
+  /**
+   * Registers an event handler on the body for a given event.
+   * @param {string} type
+   * @param {function(!Event): undefined} handler
+   */
+  registerBodyInteractionHandler(type, handler) {}
+
+  /**
+   * Deregisters an event handler on the body for a given event.
+   * @param {string} type
+   * @param {function(!Event): undefined} handler
+   */
+  deregisterBodyInteractionHandler(type, handler) {}
+
+  /**
+   * Registers an event handler for the window resize event
+   * @param {function(!Event): undefined} handler
+   */
+  registerResizeHandler(handler) {}
+
+  /**
+   * Deregisters an event handler for the window resize event
+   * @param {function(!Event): undefined} handler
+   */
+  deregisterResizeHandler(handler) {}
+
+  /**
+   * Emits a custom event MDCSlider:input from the root
+   */
+  notifyInput() {}
+
+  /**
+   * Emits a custom event MDCSlider:change from the root
+   */
+  notifyChange() {}
+
+  /**
+   * Sets a style property of the thumb container element to the passed value
+   * @param {string} propertyName
+   * @param {string} value
+   */
+  setThumbContainerStyleProperty(propertyName, value) {}
+
+  /**
+   * Sets a style property of the track element to the passed value
+   * @param {string} propertyName
+   * @param {string} value
+   */
+  setTrackStyleProperty(propertyName, value) {}
+
+  /**
+   * Sets the inner text of the pin marker to the passed value
+   * @param {number} value
+   */
+  setMarkerValue(value) {}
+
+  /**
+   * Appends the passed number of track markers to the track mark container element
+   * @param {number} numMarkers
+   */
+  appendTrackMarkers(numMarkers) {}
+
+  /**
+   * Removes all track markers fromt he track mark container element
+   */
+  removeTrackMarkers() {}
+
+  /**
+   * Sets a style property of the last track marker to the passed value
+   * @param {string} propertyName
+   * @param {string|number} value
```

Comment:
I'm not sure if our internal JS compiler will complain about `string|number` not matching the type of `setProperty`'s `value` param... The JS extern definition for `setProperty` says `value` can only be a `string`.

I guess we can try it with `string|number` for now, and if the compiler complains, we can convert `value` to `String(value)` before passing it to `setProperty`.

Reply:
I would advocate tolerating only `string` (in fact that's how it's documented inline in the foundation). It looks like we only use this API once to set `flex` so we're currently getting away with implicitly coercing number to string, but I'd be +1 on wrapping the value we're passing it with `String(...)` rather than loosening the type here.